.github/workflows/main.yml

```diff
@@ -11,8 +11,29 @@ concurrency:
   cancel-in-progress: ${{ !contains(github.ref, 'refs/heads/main') }}
 
 jobs:
+  check-skip-merge-queue:
+    name: Check if pull request can skip merge queue
+    runs-on: ubuntu-latest
+    outputs:
+      skip-merge-queue: ${{ steps.check-skip-merge-queue.outputs.up-to-date }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v6
+        if: github.event_name == 'merge_group'
```

Comment:
Nit: can we move this condition to the job level so that we don't need to repeat it twice?

Reply:
Unfortunately not, that would skip `all-jobs-pass` when not running in a merge queue.

Reply:
At least we're no longer spinning up a bunch of other VMs unnecessarily 😅